build.gradle

```diff
@@ -224,6 +224,33 @@ subprojects {
             }
         }
     }
+
+    File dexAllDir = new File(project.buildDir, "dex-all")
+    task copyDependencies(type: Copy) {
+        from configurations.runtime
+        into dexAllDir
+    }
+
+    task copyLib(type: Copy) {
+        from jar.destinationDir
+        into dexAllDir
+    }
+
+    task dexAll(type: Exec, dependsOn: [copyDependencies, copyLib]) {
+        workingDir dexAllDir
+        commandLine 'bash', '-e', '-c', """
+            ls -1 . > dexInputList.txt
+            dx.exec --dex --statistics --num-threads=4 --core-library --output=./file.dex --input-list=dexInputList.txt >dex.log 2>&1
+            grep 'method id:.*' dex.log
+        """.replace("dx.exec",(System.getProperty("dx.exec") == null) ? "dx" : System.getProperty("dx.exec"))
+    }
+
+    task methodCount(dependsOn: [build, copyDependencies, copyLib, dexAll]) {
```

Comment:
`copyDependencies, copyLib` dependencies are implied don't strictly need to be there.